libs/ui-components/src/components/Fleet/CreateFleet/steps/ConfigTemplateForm.tsx

```diff
@@ -0,0 +1,301 @@
+import * as React from 'react';
+
+import {
+  Alert,
+  Bullseye,
+  Button,
+  ExpandableSection,
+  FormGroup,
+  FormSection,
+  FormSelect,
+  FormSelectOption,
+  Grid,
+  Spinner,
+  Split,
+  SplitItem,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { FieldArray, useFormikContext } from 'formik';
+import { MinusCircleIcon } from '@patternfly/react-icons/dist/js/icons/minus-circle-icon';
+import { PlusCircleIcon } from '@patternfly/react-icons/dist/js/icons/plus-circle-icon';
+import { Repository, RepositoryList } from '@flightctl/types';
+
+import TextField from '../../../form/TextField';
+import { FleetFormValues, GitConfigTemplate, InlineConfigTemplate, KubeSecretTemplate } from '../types';
+import { useTranslation } from '../../../../hooks/useTranslation';
+import { useFetchPeriodically } from '../../../../hooks/useFetchPeriodically';
+import { getErrorMessage } from '../../../../utils/error';
+import { Trans } from 'react-i18next';
+import { Link, ROUTE } from '../../../../hooks/useNavigate';
+import TextAreaField from '../../../form/TextAreaField';
+
+const useValidateOnMount = () => {
+  const { validateForm } = useFormikContext<FleetFormValues>();
+
+  // validate new config section on mount
+  React.useEffect(() => {
+    validateForm();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+};
+
+type ConfigFormProps = {
+  index: number;
+};
+
+const GitConfigForm: React.FC<
+  ConfigFormProps & {
+    repositories: Repository[];
+  }
+> = ({ index, repositories }) => {
+  const { t } = useTranslation();
+  const { values, setFieldValue } = useFormikContext<FleetFormValues>();
+  const template = values.configTemplates[index] as GitConfigTemplate;
+
+  useValidateOnMount();
+
+  const defaultRepoName = repositories[0]?.metadata.name || '';
+
+  React.useEffect(() => {
+    if (template.repository === '') {
+      setFieldValue(`configTemplates[${index}].repository`, defaultRepoName);
+    }
+  }, [template.repository, defaultRepoName, index, setFieldValue]);
+  return (
+    <>
+      <FormGroup label={t('Repository')} isRequired>
+        <Stack hasGutter>
+          <StackItem>
+            <FormSelect
+              value={template.repository}
+              onChange={(_, value) => setFieldValue(`configTemplates[${index}].repository`, value)}
+              aria-label={t('Repository select input')}
+              isDisabled={!repositories.length}
+            >
+              {repositories.map((repo) => (
+                <FormSelectOption
+                  key={repo.metadata.name}
+                  value={repo.metadata.name}
+                  label={repo.metadata.name || ''}
+                />
+              ))}
+            </FormSelect>
+          </StackItem>
+          {repositories.length && (
```

Comment:
```suggestion
          {!repositories.length && (
```

FYI, Nir just added a new flow related to the Create fleet wizard. https://www.figma.com/file/85WfMxRxzyLEBUt3WYfzuh/Edge-management?type=design&node-id=3138-27622&mode=design&t=odDz0bzpV7nK5a7f-4

I created a separate [task](https://issues.redhat.com/browse/MGMT-17728), so for the scope of this PR it's perfect like this :+1: